sig/datadog/appsec/waf.rbs

```diff
@@ -0,0 +1,230 @@
+module Datadog
+  module AppSec
+    module WAF
+      module LibDDWAF
```

Comment:
Should this perhaps go inside libddwaf directly?

Reply:
It actually exists in libddwaf, but its `sig` isn't currently in the gem. I went the simple way first, and plan to address that ultimately.